src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -298,8 +308,7 @@ public IReadOnlyList<LoggerClass> GetLogClasses(IEnumerable<ClassDeclarationSynt
                                     }
 
                                     // ensure there are no duplicate event ids.
-                                    // LoggerMessageAttribute has constructors that don't take an EventId, we need to exclude the default Id -1 from duplication checks.
-                                    if (lm.EventId != -1 && !eventIds.Add(lm.EventId))
+                                    if (!eventIds.Add(lm.EventId))
```

Comment:
Does this mean someone could get a diagnostic because the generated ID happened to conflict?

Reply:
Ouch.

Reply:
> Does this mean someone could get a diagnostic because the generated ID happened to conflict?

IMO this is reasonable to happen. At that time users can opt-in providing their own id. This will better than just allowing the same Id for multiple logging methods. In same time I want to avoid generating different id as I want to have the generated Id value is predictable and consistent.

Reply:
@stephentoub any more thoughts here?

Reply:
I realize the failure is very unlikely to occur, but if it does occur, it will be quite confusing for the customer. Imagine if the C# compiler sometimes just failed to compile a C# source file because it has a hash collision in the names of the symbols in the type.

What else can we do?

If there's a conflict, could we just increment the id value by 1 and try again until we find an unused id?

Reply:
> If there's a conflict, could we just increment the id value by 1 and try again until we find an unused id?

This is what I am trying to avoid. This means we can generate a different Id when other parts of the code change. 

Imagine a scenario where we generate Id without any conflict and this code ships. Later in v2, more logger methods cause a conflict which causes us to regenrate the Id for the methods which previously shipped. This kind of breaking changes if anyone dependent on the Id generated in v1.

I think getting diagnostics when Id collision occurs is not that bad especially it is unlikely to happen.

Reply:
I think such a diagnostic is problematic.  We were willing to live with duplicate IDs when they defaulted to -1, I'm not sure why we're not willing to live with them when they default to something else.

I'd be fine with:
1) Just not auto-generating IDs, and if you care about IDs, assign one.
2) Auto-generating IDs but not warning when they conflict.

I don't think we should auto-generate IDs and warn when they conflict.  Then harmless activities like tweaking a logging method name can start producing new seemingly unrelated and haphazard warnings.

Reply:
Ok, I applied #2 in the commit [`b1d5a89` (#87892)](https://github.com/dotnet/runtime/pull/87892/commits/b1d5a89bb42e590d62dce074eae0f0b618155d32)